crates/pprof/src/lib.rs

```diff
@@ -1,5 +1,38 @@
+//! A crate designed to provide a more ergonomic interface to the `pprof` crate.
+//!
+//! ## Example
+//!
+//! ```rust,no_run
+//! // Create a new CPU profiler with a sampling frequency of 1000 Hz and an empty ignore list.
+//! let cpu = scuffle_pprof::Cpu::new::<String>(1000, &[]);
+//!
+//! // Capture a pprof profile for 10 seconds.
+//! // This call is blocking. It is recommended to run it in a separate thread.
+//! let capture = cpu.capture(std::time::Duration::from_secs(10)).unwrap();
+//!
+//! // Write the profile to a file.
+//! std::fs::write("capture.pprof", capture).unwrap();
```

Comment:
Perhaps also show some `go pprof` tool commands for generating a flame graph of the capture